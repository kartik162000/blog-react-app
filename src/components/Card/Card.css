.main {
    flex: 1;
}

.card-main {
    width: 300px;
    height: 400px;
    background-color: #fff;
    margin: 10px 0px;
    border: 0.5px solid rgb(156, 156, 156);
}

.card-image{
    width: 100%;
    height: 60%;
}

.card-image img {
    width: 100%;
    height: 60%;
    background-size: cover;
    background-repeat: no-repeat;
    background-position: center;
    flex-shrink: 1;
    min-width: 100%;
    min-height: 100%
}
.card-content
{
    padding: 0px 5px;
    padding-top: 5px;
    height: 40%;
}
.card-caption
    {
        height: 15%;
        width: 100%;
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        font-size: 0.75rem;
    font-family: 'Roboto', sans-serif;
    color: rgb(137, 137, 137);
    }

.card-title {
    height: 30%;
    /* padding: 0px 6px; */
    width: 100%;
}
.card-title h4
{
    margin: 0;
    padding: 0;
    font-size: auto;
}

.card-description p {
    margin: 0;
    /* padding-top: 10px; */
    font-size: 12px;
    font-weight: 100;
    font-family: 'Roboto', sans-serif;
    color: rgb(104, 104, 104);

}

#cardLine {
    width: 100%;
    border: 0.1px solid rgb(0, 0, 0, 0.2);
}

.card-footer {
    height: 14%;
    width: 100%;
    display: flex;
    justify-content: space-between;
}

.card-clap img {
    height: 80%;
    margin: 0px 10px;
}

.card-clap-count
{
    display: inline-block;
    width: 50px;
    color: rgb(137, 137, 137);
} 
.card-heart img
{
    height: 80%;
    margin: 0px 10px;
}