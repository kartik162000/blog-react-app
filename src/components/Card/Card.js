import React from "react";
import clapIcon from "/Users/kartik_goel/Desktop/front_end week/blog-post/src/assets/Icons/clapping.svg";
import heartRedIcon from "/Users/kartik_goel/Desktop/front_end week/blog-post/src/assets/Icons/heart-red.svg";
import heartBlankIcon from "/Users/kartik_goel/Desktop/front_end week/blog-post/src/assets/Icons/heart-black.svg";
import { useState } from "react";

import "./Card.css";
function Card({ date, readingTime, title, description, claps, liked, image }) {
  const [countClap, setCountClap] = useState(claps);
  const [isLiked, setIsLiked] = useState(liked);
  const clickHandler = () => {
    setCountClap(countClap + 1);
  };

  const heartClickHandler = () => {
    setIsLiked(!isLiked);
  };

  return (
    <div className="card-main">
      <div className="card-image">
        <img src={require(`../../assets/images/${image}`)} alt="image" />
      </div>
      <div className="card-content">
        <div className="card-caption">
          <div className="card-date">{date}</div>
          <div className="card-reading-time">{readingTime}</div>
        </div>
        <div className="card-title">
          <h4>{title}</h4>
        </div>
        <div className="card-description">
          <p>{description}</p>
        </div>
        <hr id="cardLine"></hr>
        <div className="card-footer">
          <div className="card-clap">
            <img src={clapIcon} alt="clap" onClick={clickHandler} />
            <div className="card-clap-count">{countClap}</div>
          </div>
          <div className="card-heart">
            {isLiked ? (
              <img src={heartRedIcon} alt="heart" onClick={heartClickHandler} />
            ) : (
              <img
                src={heartBlankIcon}
                alt="heart"
                onClick={heartClickHandler}
              />
            )}
          </div>
        </div>
      </div>
    </div>
  );
}

export default Card;
