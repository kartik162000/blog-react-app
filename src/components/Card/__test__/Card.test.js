import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import Card from "../Card";

describe("Card Component rendring test", () => {
  const mockProps = {
    id: 1,
    date: "2021-08-01",
    readingTime: "5 min read",
    title: "How to write a test",
    description: "This is a test description",
    claps: 10,
    liked: false,
    image: "abstract.png",
  };
  it("should render when component is mounted", () => {
    const { asFragment } = render(<Card {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("should render the correct date", () => {
    const { getByText } = render(<Card {...mockProps} />);
    expect(getByText("2021-08-01")).toBeInTheDocument();
  });
  it("should render with the increase count", () => {
    const { container } = render(<Card {...mockProps} />);
    const clapButton = screen.getByAltText("clap");
    const text = container.querySelector(".card-clap-count");
    expect(text.innerHTML).toContain("10");
    fireEvent.click(clapButton);
    expect(text.innerHTML).toContain("11");
  });
  it("should render with the correct heart icon when it is clicked", () => {
    render(<Card {...mockProps} />);
    const heartButton = screen.getByAltText("heart");
    expect(heartButton.src).toContain("heart-black.svg");
    fireEvent.click(heartButton);
    expect(heartButton.src).toContain("heart-red.svg");
  });
});
