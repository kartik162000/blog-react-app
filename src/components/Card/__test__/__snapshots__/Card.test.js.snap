// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component rendring test should render when component is mounted 1`] = `
<DocumentFragment>
  <div
    class="card-main"
  >
    <div
      class="card-image"
    >
      <img
        alt="image"
        src="abstract.png"
      />
    </div>
    <div
      class="card-content"
    >
      <div
        class="card-caption"
      >
        <div
          class="card-date"
        >
          2021-08-01
        </div>
        <div
          class="card-reading-time"
        >
          5 min read
        </div>
      </div>
      <div
        class="card-title"
      >
        <h4>
          How to write a test
        </h4>
      </div>
      <div
        class="card-description"
      >
        <p>
          This is a test description
        </p>
      </div>
      <hr
        id="cardLine"
      />
      <div
        class="card-footer"
      >
        <div
          class="card-clap"
        >
          <img
            alt="clap"
            src="clapping.svg"
          />
          <div
            class="card-clap-count"
          >
            10
          </div>
        </div>
        <div
          class="card-heart"
        >
          <img
            alt="heart"
            src="heart-black.svg"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
